lib/js/src/manager/file/BaseFileManager.js

```diff
@@ -0,0 +1,245 @@
+        const rpcRequest = new RpcRequest(
+            {
+                functionName: 'DeleteFile',
+                parameters: {
+                    syncFileName: fileName,
+                },
+                rpcType: RpcType.REQUEST,
+            }
+        );
+
+        const rpcResponse = await this._internalInterface.sendRPC(rpcRequest);
+        const isSuccess = rpcResponse.getSuccess();
+        if (isSuccess) {
+            this._bytesAvailable = rpcResponse.getSpaceAvailable() ? rpcResponse.getSpaceAvailable() : SPACE_AVAILABLE_MAX_VALUE;
+            this._remoteFiles.splice(this._remoteFiles.indexOf(fileName), 1);
+            this._uploadedEphemeralFileNames.splice(this._remoteFiles.indexOf(fileName), 1);
+        }
+        if (onComplete !== null) {
+            onComplete(isSuccess);
+        }
+        return isSuccess;
+    }
+
+
+    /**
+	 * Attempts to delete a list of files from core, calls onComplete with indication of success/failure.
+	 * @param fileNames list of file names to be deleted
+	 * @param onComplete callback that is called once core responds to all deletion requests
+	 */
+    async deleteRemoteFilesWithNames (fileNames, onComplete = null) {
+        const results = [];
+
+        if (fileNames.length > 0) {
+            for (const filename of fileNames) {
+                results.push(this.deleteRemoteFileWithName(filename));
+            }
+            for (const idx in results) {
+                results[idx] = await Promise.resolve(results[idx]);
+            }
+        }
+
+        if (onComplete !== null) {
+            onComplete(results);
+        }
+        return results;
+    }
+
+    async uploadFile (file, onComplete = null) {
+        if (file.isStaticIcon()) {
+            // Static icons don't need to be uploaded
+            onComplete(true);
+            return;
+        }
+        const putFile = this.createPutFile(file);
+
+        const putFileResponse = await this._internalInterface.sendRPC(putFile);
+        if (putFileResponse.getSuccess()) {
+            this._bytesAvailable = putFileResponse.getSpaceAvailable() !== null ? putFileResponse.getSpaceAvailable() : SPACE_AVAILABLE_MAX_VALUE;
+            this._remoteFiles.push(file.getName());
+            this._uploadedEphemeralFileNames.push(file.getName());
+        }
+        if (onComplete !== null) {
+            onComplete(putFileResponse.getSuccess());
+        }
+
+        return putFileResponse.getSuccess();
+    }
+
+    /**
+	 * Attempts to upload a list of SdlFiles to core
+	 * @param files list of SdlFiles with file name and one of A) fileData, B) Uri, or C) resourceID set
+	 * @param listener callback that is called once core responds to all upload requests
+	 */
+    async uploadFiles (files, onComplete = null) {
+        const results = [];
+        if (files.length > 0) {
+            for (const file of files) {
+                results.push(this.uploadFile(file));
+            }
```

Comment:
Skeptical of whether or not this would execute the PutFile RPCs asynchronously like the Java Suite does. Investigation required.